addons/mail/static/src/messaging/service/messaging_service/messaging_service.js

```diff
@@ -1,50 +1,42 @@
 odoo.define('mail.messaging.service.Messaging', function (require) {
 'use strict';
 
-const {
-    checkRelations,
-    generateEntities,
-} = require('mail.messaging.entity.core');
+const { addMessagingToEnv } = require('mail.messaging.messaging_env');
 
 const AbstractService = require('web.AbstractService');
 const { serviceRegistry } = require('web.core');
 const env = require('web.env');
 
-const { Store } = owl;
-const { EventBus } = owl.core;
+addMessagingToEnv(env);
 
 const MessagingService = AbstractService.extend({
     env,
-    messagingEnvExtension: undefined,
     /**
      * @override {web.AbstractService}
      */
     start() {
         this._super(...arguments);
-        window.addEventListener('load', () => this._onGlobalLoad());
 
-        /**
-         * Environment of the messaging store (messaging env. without store)
-         */
         Object.assign(this.env, {
-            autofetchPartnerImStatus: true,
-            disableAnimation: false,
             call: (...args) => this.call(...args),
             do_action: (...args) => this.do_action(...args),
             do_notify: (...args) => this.do_notify(...args),
             do_warn: (...args) => this.do_warn(...args),
-            entities: undefined,
-            isMessagingInitialized() {
-                if (!this.messaging) {
-                    return false;
-                }
-                return this.messaging.isInitialized;
-            },
-            messaging: undefined,
-            messagingBus: new EventBus(),
             rpc: (...args) => this._rpc(...args),
             trigger_up: (...args) => this.trigger_up(...args),
-        }, this.messagingEnvExtension);
+        });
+
+        /**
+         * Messaging initialization.
+         */
+        const messagingInitializedPromise = this.env.messagingCreatedPromise.then(async () => {
+            // TODO FIXME The method uses service specific env keys so it can
+            // only be called after a service has properly set up those keys.
+            await this.env.messaging.start();
```

Comment:
So if I get it right, you need this because of uses of `trigger_up`/`call`/`rpc`/`do_action`/etc. a bit everywhere in code.

This is a bit hard to remove these legacy processes mainly because of mock server, and I assume because of this you may still need messaging service in front-end with livechat, as messaging entity is essential for the good working of messaging env.

I don't have a solution for this issue yet, and definitely none for the scope of this branch.
Still racking my brain on that from time to time, unfortunately not the highest priority for the moment as entity and concurrency issues are a bigger concern.

Reply:
Well yes, this is the whole reason why we need messaging service as you explained to me.
We could get rid of the service it wasn't for those extra methods of legacy code.

We will need a public messaging service too for the same reason, which will be using the messaging_env the same way, except the default env will be the public one instead.